docker.go

```diff
@@ -177,23 +177,23 @@ func (p Plugin) Exec() error {
 	}
 
 	// add base image docker credentials to the existing config file, else create new
+	// instead of writing to config file directly, using docker's login func
```

Comment:
can you also explain what's the difference and why one is preferred to another?

Reply:
The login function integrates with various credential helpers, it also handles different registry specific logic.